src/CTA.Rules.Actions/ClassActions.cs

```diff
@@ -152,5 +158,27 @@ public class ClassActions
             };
             return RenameClass;
         }
+        public Func<SyntaxGenerator, ClassDeclarationSyntax, ClassDeclarationSyntax> GetReplaceMethodModifiersAction(string methodName, string modifiers)
+        {
+            Func<SyntaxGenerator, ClassDeclarationSyntax, ClassDeclarationSyntax> ReplaceMethodModifiers = (SyntaxGenerator syntaxGenerator, ClassDeclarationSyntax node) =>
+            {
+                var allMembers = node.Members.ToList();
+                var allMethods = allMembers.OfType<MethodDeclarationSyntax>();
+                if(allMethods.Count() > 0)
+                {
+                    var replaceMethod = allMethods.Where(m => m.Identifier.ToString() == methodName);
+                    if(replaceMethod.Count() > 0)
+                    {
+                        SyntaxTokenList tokenList = new SyntaxTokenList(SyntaxFactory.ParseTokens(modifiers));
+                        var newMethod = replaceMethod.FirstOrDefault().WithModifiers(tokenList);
+
+                        node = node.WithMembers(node.Members.Replace(replaceMethod.FirstOrDefault(), newMethod)).NormalizeWhitespace();
+                    }
```

Comment:
nit: When getting replaceMethod, use FirstOrDefault, then check if it's null. This way you won't have to call FirstOrDefault twice inside the condition.

Reply:
Re-tested as mentioned and updated.